Orm/Xtensive.Orm/Reflection/TypeHelper.cs

```diff
@@ -592,38 +609,40 @@ public static object Activate(this Type type, Type[] genericArguments, params ob
     }
 
     /// <summary>
-    /// Gets the public constructor of type <paramref name="type"/> 
-    /// accepting specified <paramref name="arguments"/>.
+    /// Gets the public constructor of type <paramref name="type"/>
+    /// accepting specified <paramref name="argumentTypes"/>.
     /// </summary>
     /// <param name="type">The type to get the constructor for.</param>
-    /// <param name="arguments">The arguments.</param>
+    /// <param name="argumentTypes">The argument types.</param>
     /// <returns>
     /// Appropriate constructor, if a single match is found;
     /// otherwise, <see langword="null"/>.
     /// </returns>
-    public static ConstructorInfo GetConstructor(this Type type, object[] arguments)
-    {
+    public static ConstructorInfo GetSingleConstructor(this Type type, Type[] argumentTypes) =>
+      constructorInfoByTypes.GetOrAdd((type, argumentTypes), ConstructorExtractor);
```

Comment:
Since the class is public I can't guarantee that no one except DataObjects.Net uses it. Renaming of publicly available members should have pretty strong reasons. Why did you rename the method? I don't see any reasons.

Reply:
I changed name because of ambiguity with method `Type.GetConstructor(Type[])`
https://docs.microsoft.com/en-us/dotnet/api/system.type.getconstructor?view=net-5.0#System_Type_GetConstructor_System_Type___

Note, argument type was changed.

Now returned old method extension as well, for backward compatibility.

Reply:
I prefer marking method [Obsolete] and provide new alternative. It makes transition not so hart. If you renamed the method this way I would not say anything.

Reply:
But now there are two variants and both have no ambiguity, right?